packages/faustwp-core/src/components/Toolbar/Toolbar.tsx

```diff
@@ -0,0 +1,183 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { Edit } from './nodes/Edit.js';
+import { GraphiQL } from './nodes/GraphiQL.js';
+import { MyAccount } from './nodes/MyAccount.js';
+import { hooks } from '../../wpHooks/index.js';
+import { ToolbarNode } from './ToolbarNode.js';
+import { SeedNode } from '../../queries/seedQuery.js';
+import { useAuth } from '../../hooks/useAuth.js';
+import { SiteName } from './nodes/SiteName.js';
+
+/**
+ * The available menu locations that nodes can be added to.
+ */
+export type MenuLocation = 'primary' | 'secondary';
+
+/**
+ * A single Faust Toolbar node.
+ */
+export type FaustToolbarNode = {
+  /**
+   * The identifier for this Toolbar Node.
+   * Used to create each Toolbar Node's id. `#wp-admin-bar-{id}`
+   */
+  id: string;
+
+  /**
+   * The available menu locations that nodes can be added to.
+   */
+  location: MenuLocation;
+
+  /**
+   * The JSX component to load.
+   * This will be rendered inside of an `<li>`.
+   * @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/li
+   */
+  component: React.ReactElement;
+
+  /**
+   * Additional classNames for the Toolbar Node `<li></li>`.
+   */
+  additionalClassNames?: string;
+};
+
+/**
+ * An array of toolbar nodes.
+ */
+export type FaustToolbarNodes = FaustToolbarNode[];
+
+/**
+ * Toolbar context.
+ */
+export type FaustToolbarContext = {
+  seedNode?: SeedNode;
+};
+
+/**
+ * Toolbar props.
+ */
+export type ToolbarProps = {
+  seedNode?: SeedNode;
+};
+
+/**
+ * Renders a Toolbar that is based on WordPress' own toolbar.
+ */
+export function Toolbar({ seedNode }: ToolbarProps) {
+  /**
+   * Define Toolbar Nodes that should be included by default.
+   */
+  const coreToolbarNodes: FaustToolbarNodes = useMemo(() => {
+    return [
+      {
+        id: 'site-name',
+        location: 'primary',
+        component: <SiteName />,
+      },
+      {
+        id: 'edit',
+        location: 'primary',
+        component: <Edit seedNode={seedNode} />,
+      },
+      {
+        id: 'graphiql',
+        location: 'primary',
+        component: <GraphiQL />,
+      },
+      {
+        id: 'my-account',
+        location: 'secondary',
+        component: <MyAccount />,
+        additionalClassNames: 'with-avatar',
+      },
+    ];
+  }, [seedNode]);
+
+  const [toolbarNodes, setToolbarNodes] = useState(coreToolbarNodes);
+  let { isAuthenticated } = useAuth();
+
+  isAuthenticated = true;
```

Comment:
Possible remnant of testing that should be removed

Reply:
Yes, thank you!